events/janus_wsevh.c

```diff
@@ -839,3 +828,53 @@ static int janus_wsevh_callback(struct lws *wsi, enum lws_callback_reasons reaso
 	}
 	return 0;
 }
+
+/**
+ * Implements the connecting attempt to the upstream websocket server
+ * sets the connection result (lws_client_connect_info) to static wsi
+ */
+static void janus_wsevh_connect_attempt(void* sul) {
+	struct lws_client_connect_info i = { 0 };
+	i.host = address;
+	i.origin = address;
+	i.address = address;
+	i.port = port;
+	i.path = path;
+	i.context = context;
+	if(!strcasecmp(protocol, "wss"))
+		i.ssl_connection = 1;
+	i.ietf_version_or_minus_one = -1;
+	i.client_exts = exts;
+	i.protocol = protocols[0].name;
+	JANUS_LOG(LOG_INFO, "WebSocketsEventHandler: Connecting to upstream websocket server %s:%d...\n", address, port);
+	wsi = lws_client_connect_via_info(&i);
+	if(!wsi) // As we specified a callback pointer in the context the NULL result is unlikely to happen -> just logging it here in case we see it in the field
+		JANUS_LOG(LOG_ERR, "WebSocketsEventHandler: Connecting to upstream websocket server %s:%d failed\n", address, port);
+	reconnect = FALSE;
+}
+
+/**
+ * Adopts the reconnect_delay value in case of an error
+ * Increases the value up to JANUS_WSEVH_MAX_RETRY_SECS
+ */
+void janus_wsevh_calculate_reconnect_delay_on_fail(void) {
+	if(reconnect_delay == 0)
+		reconnect_delay = 1;
+	else if (reconnect_delay < JANUS_WSEVH_MAX_RETRY_SECS) {
+		reconnect_delay += reconnect_delay;
+		if (reconnect_delay > JANUS_WSEVH_MAX_RETRY_SECS)
+			reconnect_delay = JANUS_WSEVH_MAX_RETRY_SECS;
+	}
+	disconnected = janus_get_monotonic_time();
+	reconnect = TRUE;
+	JANUS_LOG(LOG_WARN, "WebSocketsEventHandler: Error attempting connection... (next retry in %ds)\n", reconnect_delay);
```

Comment:
imho it's not intuitive to change connection status in a method used te calculate the re-connection delay
I'd move `disconnected` assignment, `reconnect` assignment and the log about the error outside of the method (sometimes duplicating the code makes it more readable)

Reply:
i duplicated it and placed it accordingly